src/sec_edgar_bulker/core/proxy_manager.py

```
@@ -0,0 +1,256 @@
+        """
+        self.config = config
+        self.settings = config.proxy
+        self.mode = self.settings.mode
+        self.proxies: List[Tuple[str, Optional[aiohttp.BasicAuth]]] = []
+        self.proxy_usage_count: Dict[str, int] = {}
+        self.used_proxies: Set[str] = set()  # Track used proxies for rotation
+        self.logger = logging.getLogger("sec_edgar_bulker.proxy")
+        
+        # Always check required state first
+        if self.settings.required:
+            if not self.settings.enabled:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)